streaming/src/main/scala/org/apache/gearpump/streaming/dsl/window/api/WindowFunction.scala

```diff
@@ -33,11 +33,19 @@ object WindowFunction {
 }
 
 trait WindowFunction[T] {
+
   def apply(context: WindowFunction.Context[T]): Array[Window]
+
+  def isNonMerging: Boolean
+}
+
+abstract class NonMergingWindowFunction[T] extends WindowFunction[T] {
+
+  override def isNonMerging: Boolean = true
```

Comment:
But if user's WindowFunction is not able to be merged, extending `NonMergingWindowFunction` would suffice, otherwise extending `WindowFunction`